### ISSUE TITLE
TOC doesn't show header which includes formulas correctly

### ISSUE THREAD

**Issue by JingwangLi** [Python-Markdown/markdown#895]
I found TOC doesn't show header which includes formulas correctly even the header is normal in the body.
![image](https://user-images.githubusercontent.com/29228612/72410217-de2de200-37a2-11ea-80da-5c9437ec8d59.png)
![image](https://user-images.githubusercontent.com/29228612/72410231-e4bc5980-37a2-11ea-9825-4608ca300898.png)

How to fix this?

**Comment by mitya57** [comment]
You may need to add a `data-toc-label` attribute like described in https://github.com/Python-Markdown/markdown/issues/699#issuecomment-413964476. See also the other discussion in #699 for more details.

Also, you have not specified what math extension you are using.

**Comment by JingwangLi** [comment]
> You may need to add a `data-toc-label` attribute like described in [#699 (comment)](https://github.com/Python-Markdown/markdown/issues/699#issuecomment-413964476). See also the other discussion in #699 for more details.
> 
> Also, you have not specified what math extension you are using.

Thank you. I have seen the comment and the docs of Custom Labels. But I think the usage of `data-toc-label` is not specific and it makes me confused,  so I just tried `# $a_i$ { #a_i data-toc-label='<script type="math/tex">$a_i$</script>' }` which  appeared in the comment, but it doesn't work:
![image](https://user-images.githubusercontent.com/29228612/72434039-a1c7a980-37d5-11ea-9f32-b7eb76cd59e9.png)
![image](https://user-images.githubusercontent.com/29228612/72434074-b86e0080-37d5-11ea-9f1c-f01ad8b47f43.png)

Actually I'm using the [Markdown Preview](https://github.com/facelessuser/MarkdownPreview/issues/107) which uses Python-Markdown. I found this problem when I'm using Markdown Preview, but the author of Markdown Preview said that's Python-Markdown's problem so I moved the issue here. Therefore I don't know what math extension Markdown Preview is using exactly.

And, I found a Markdown rendering engine(in Hexo) I used shows header with math normally without external attribute, maybe there's something we can learn from it.
![image](https://user-images.githubusercontent.com/29228612/72435118-4f3bbc80-37d8-11ea-8f66-b9078a085cfe.png)

**Comment by mitya57** [comment]
> I just tried `# $a_i$ { #a_i data-toc-label='<script type="math/tex">$a_i$</script>' }` which  appeared in the comment, but it doesn't work

- Please make sure you have the [`attr_list` extension](https://python-markdown.github.io/extensions/attr_list/) enabled.
- Remove the dollar signs inside the `<script>` tag.

**Comment by facelessuser** [comment]
I can confirm that the following works:

```
[TOC]

# $a_i$ { #a_i data-toc-label='<script type="math/tex">a_i</script>' }
```

I had never tried that in Arithmatex, but it apparently works just fine.

**Comment by bradleyharden** [comment]
Sorry to dig this issue up, but I can't seem to get it to work with the example given by @facelessuser. I also don't know if my issue is with MkDocs or with the extension. I'm probably making a simple mistake, but I just don't know much about HTML or the ecosystem, so I'm having trouble finding a lead to follow.

I created a minimal reproducer [here](https://github.com/bradleyharden/mkdoc_math_in_toc). Would it be possible for someone to take a look?

Edit: I realized it might not be clear what my problem is. I can get $a_i$ to render correctly within the contents of the page, but it still shows up as `a_i` in the TOC. I stripped it down to the bare minimum, but I still can't get it to work, and I don't know why.
